bittensor/_synapse/text_lasthiddenstate_impl.py

```diff
@@ -111,27 +143,88 @@ def check_forward_response_tensor    ( self, forward_request_tensor, forward_res
         if forward_response_tensor == None:
             raise ValueError('Empty Response')
             
-        if ( 
-             len( forward_response_tensor.shape ) != 3 or
-             forward_response_tensor.size(0) != forward_request_tensor.size(0) or
-             forward_response_tensor.size(1) != forward_request_tensor.size(1) or
-             forward_response_tensor.size(2) != bittensor.__network_dim__
-            ):
-            raise ValueError( "forward_response_tensor.shape must be in [{}, {}, {}], got: {} for synapse: {}".format( forward_request_tensor.size(0) , forward_request_tensor.size(1), bittensor.__network_dim__, list(forward_response_tensor.shape), self ) ) 
-   
+        if len( forward_response_tensor.shape ) == 3:
+            if ( forward_response_tensor.size(0) != forward_request_tensor.size(0) or
+                 forward_response_tensor.size(1) != forward_request_tensor.size(1) or
+                 forward_response_tensor.size(2) != bittensor.__network_dim__ ):
+                raise ValueError( "forward_response_tensor.shape must be in [{}, {}, {}], got: {} for synapse: {}".format( forward_request_tensor.size(0) , forward_request_tensor.size(1), bittensor.__network_dim__, list(forward_response_tensor.shape), self ) ) 
+        
+        elif len( forward_response_tensor.shape ) == 2:
+            mask_len = forward_request_tensor.size(1) if self.mask == [] else len(self.mask)
+            if ( forward_response_tensor.size(0) != forward_request_tensor.size(0) * mask_len or
```

Comment:
Is the response tensor (batch*mask_len, network_dim)? Any reason why we want to push it down to 2 ranks instead of the (batch, mask_len, network_dim)?

Reply:
On a similar note, there should a check if the mask > than the size of the request tensor here